lib/tests/test_ecl_grid.cpp

```diff
@@ -0,0 +1,298 @@
+      ecl_kw_iset_float(coord_kw, offset + 1, j);
+      ecl_kw_iset_float(coord_kw, offset + 2, -1);
+
+      ecl_kw_iset_float(coord_kw, offset + 3, i);
+      ecl_kw_iset_float(coord_kw, offset + 4, j);
+      ecl_kw_iset_float(coord_kw, offset + 5, -1);
+
+      for (int k = 0; k < num_z; k++) {
+        for (int c = 0; c < 4; c++) {
+          int zi1 = ecl_grid_zcorn_index__(num_x, num_y, i, j, k, c);
+          int zi2 = ecl_grid_zcorn_index__(num_x, num_y, i, j, k, c + 4);
+
+          double z1 = k;
+          double z2 = k + 1;
+
+          ecl_kw_iset_float(zcorn_kw, zi1, z1);
+          ecl_kw_iset_float(zcorn_kw, zi2, z2);
+        }
+      }
+    }
+  }
+
+  for (auto &t : z_vector) {
+    int i, j, k, c;
+    double z;
+    tie(i, j, k, c, z) = move(t);
+    auto index = ecl_grid_zcorn_index__(num_x, num_y, i, j, k, c);
+    ecl_kw_iset_float(zcorn_kw, index, z);
+  }
+
+  ecl_grid_type *grid = ecl_grid_alloc_GRDECL_kw(num_x, num_y, num_z, zcorn_kw,
+                                                 coord_kw, NULL, NULL);
+  ecl_kw_free(coord_kw);
+  ecl_kw_free(zcorn_kw);
+
+  return grid;
+}
+
+TEST_CASE("Test no twistedness", "[unittest]") {
+  GIVEN("A grid with untwisted cells") {
+
+    ecl_grid_type *grid = generate_coordkw_grid(
+        10, 7, 8, vector<tuple<int, int, int, int, double>>(0));
+
+    AND_GIVEN("Any cell in that grid") {
+      auto i = GENERATE(take(3, random(0, 9)));
+      auto j = GENERATE(take(3, random(0, 6)));
+      auto k = GENERATE(take(3, random(0, 7)));
+
+      THEN("That cell should have a twist score of 0") {
+        REQUIRE(ecl_grid_get_cell_twist3(grid, i, j, k) == 0);
+      }
+    }
+
+    ecl_grid_free(grid);
+  }
+}
+
+TEST_CASE("Test simple twistedness", "[unittest]") {
+  GIVEN("A grid with the first cell twisted") {
+
+    auto z_vector = vector<tuple<int, int, int, int, double>>(2);
+    z_vector[0] = make_tuple(0, 0, 0, 0, 0.0);
+    z_vector[1] = make_tuple(0, 0, 0, 4, -0.25);
+    ecl_grid_type *grid = generate_coordkw_grid(10, 7, 8, z_vector);
+
+    THEN("That cell should have a twist score of 1") {
+      REQUIRE(ecl_grid_get_cell_twist3(grid, 0, 0, 0) == 1);
+    }
+
+    ecl_grid_free(grid);
+  }
+}
+
+TEST_CASE("Test double twistedness", "[unittest]") {
+  GIVEN("A grid with the first cell doubly twisted") {
+
+    auto z_vector = vector<tuple<int, int, int, int, double>>(4);
+    z_vector[0] = make_tuple(0, 0, 0, 0, 0.0);
+    z_vector[1] = make_tuple(0, 0, 0, 4, -0.25);
+    z_vector[2] = make_tuple(0, 0, 0, 3, 0.0);
+    z_vector[3] = make_tuple(0, 0, 0, 7, -0.25);
+    ecl_grid_type *grid = generate_coordkw_grid(10, 7, 8, z_vector);
+
+    THEN("That cell should have a twist score of 2") {
+      REQUIRE(ecl_grid_get_cell_twist3(grid, 0, 0, 0) == 2);
+    }
+
+    ecl_grid_free(grid);
+  }
+}
+
+class Tmpdir {
+    public:
+        fs::path dirname;
+        Tmpdir() : dirname("") {
+
+            auto parentdir = fs::temp_directory_path() / "ecl_test";
```

Comment:
Output directory to `stderr` to notify the user about the new created tmpdir.

Reply:
Prints directory to stderr.